src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1253,6 +1260,136 @@ def get_enable_ahub(self, enable_validation: bool = False, **kwargs):
 
+    # pylint: disable=unused-argument
+    def get_enable_managed_identity_service_principal_and_client_secret(
+        self, enable_validation: bool = False, **kwargs
+    ):
+        # service_principal
+        # read the original value passed by the command
+        sp_raw_value = self.raw_param.get("service_principal")
+        # try to read from intermediates
+        sp_intermediate = self.get_intermediate("service_principal", None)
+        # try to read the property value corresponding to the parameter from the `mc` object
+        sp_value_obtained_from_mc = None
+        if self.mc and self.mc.service_principal_profile:
+            sp_value_obtained_from_mc = (
+                self.mc.service_principal_profile.client_id
+            )
+
+        # set default value
+        sp_read_from_mc = False
+        if sp_value_obtained_from_mc is not None:
+            service_principal = sp_value_obtained_from_mc
+            sp_read_from_mc = True
+        elif sp_intermediate is not None:
+            service_principal = sp_intermediate
+        else:
+            service_principal = sp_raw_value
+
+        # client_secret
+        # read the original value passed by the command
+        secret_raw_value = self.raw_param.get("client_secret")
+        # try to read from intermediates
+        secret_intermediate = self.get_intermediate("client_secret", None)
+        # try to read the property value corresponding to the parameter from the `mc` object
+        secret_value_obtained_from_mc = None
+        if self.mc and self.mc.service_principal_profile:
+            secret_value_obtained_from_mc = (
+                self.mc.service_principal_profile.secret
+            )
+
+        # set default value
+        secret_read_from_mc = False
+        if secret_value_obtained_from_mc is not None:
+            client_secret = secret_value_obtained_from_mc
+            secret_read_from_mc = True
+        elif secret_intermediate is not None:
+            client_secret = secret_intermediate
+        else:
+            client_secret = secret_raw_value
+
+        # enable_managed_identity
+        # Note: This parameter will not be decorated into the `mc` object.
+        # read the original value passed by the command
+        managed_identity_raw_value = self.raw_param.get(
+            "enable_managed_identity"
+        )
+        # try to read from intermediates
+        managed_identity_intermediate = self.get_intermediate(
+            "enable_managed_identity", None
+        )
+
+        # set default value
+        if managed_identity_intermediate is not None:
+            enable_managed_identity = managed_identity_intermediate
+        else:
+            enable_managed_identity = managed_identity_raw_value
+
+        # dynamic completion for enable_managed_identity
+        if service_principal and client_secret:
+            enable_managed_identity = False
+            # add to intermediate
+            self.set_intermediate(
```

Comment:
looks like this intermediate values are unnecessary.
we can put a lock on the mc or so to remove this.

Reply:
Good point, changed in commit [#1cac795](https://github.com/Azure/azure-cli/pull/19353/commits/1cac7959eca08e5208a12b09b9aa44e9f1e01c50).